src/arch/x86/lib/cpu_performance/cpu_info.c

```diff
@@ -12,7 +12,37 @@
 
 struct cpu_info current_cpu;
 
+static void set_cpu_features(struct cpu_info *info) {
+	uint32_t reg[1]; /* register: edx */
+
+	asm volatile (	"movl $0x1, %%eax   \n\t"
+					"cpuid				\n\t"
+					"movl %%edx, %0		\n\t"
+					: "=g"(reg[0])
+					:
+					: "%eax", "%ebx", "%ecx", "%edx");
+
+	info->FPU = (reg[0] << 31) >> 31;
+	info->MMX = (reg[0] << 8) >> 31; 
+	info->SSE = (reg[0] << 6) >> 31;
```

Comment:
I think it will be more readable if you simplify it as `sse = (reg >> 25) & 0x1`, this way reader can instantly see that you want to get 25-th bit of the register.